lib/scala/kg/src/main/scala/formats/kg/kgtk/KgtkTsvReader.scala

```diff
@@ -0,0 +1,71 @@
+package formats.kg.kgtk
+
+import formats.CsvReader
+import com.github.tototoshi.csv.{CSVReader, TSVFormat}
+import models.kg.KgEdge
+import models.kg.KgNode
+import org.slf4j.LoggerFactory
+import java.nio.file.Path
+import java.io.InputStream
+
+final class KgtkTsvReader(csvReader: CSVReader) extends CsvReader[Tuple3[KgEdge, KgNode, KgNode]](csvReader) {
+  private final val LIST_DELIM = "|";
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  def iterator: Iterator[Tuple3[KgEdge, KgNode, KgNode]] =
+    csvReader.iteratorWithHeaders.map(row => {
+      val node1Labels = row.getList("node1;label", LIST_DELIM)
+      val node2Labels = row.getList("node2;label", LIST_DELIM)
+      Tuple3(
+        KgEdge(
+          datasource = row("datasource"),
+          datasources = row.getList("datasource", "|"),
+          id = row("id"),
+          `object` = row("node2"),
```

Comment:
Here's another case where I'd make the domain model be a non-redundant representation (i.e., only the fields in the Kgtk model) and adapt at the boundaries:
* the GraphQL API, fill in fields that used to exist so the front end is not broken
* the ingestion side e.g., the CsvReader subclasses, read old fields and translate them

This is a standard migration technique.

I was wrong yesterday, and/or not explicit enough.

Reply:
Clarified in slack, updated models